File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        checked_dict = {}
        for i,n in enumerate(nums):
            if target-n in checked_dict:
                return [checked_dict[target-n],i]
            else:
                checked_dict[n] = i
            

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash; //hash: number->index
        for(int i=0; i<nums.size(); ++i){
            int complement = target-nums[i];
            if (hash.count(complement)){
                return {hash[complement],i};
            }else{
                hash[nums[i]]=i;
            }
        }
        return {};
    }
};